tembo-cli/src/main.rs

```diff
@@ -1,112 +1,60 @@
-#[macro_use]
-extern crate clap;
-extern crate log;
-extern crate serde;
-extern crate serde_yaml;
-extern crate simplelog;
+use clap::{Args, crate_authors, crate_version, Parser, Subcommand};
+use cmd::apply::ApplyCommand;
+use cmd::context::{ContextCommand, ContextSubCommand};
+use cmd::init::InitCommand;
 
-use anyhow::Context;
-use clap::{Arg, Command};
-use clap_complete::Shell;
-use serde::{Deserialize, Serialize};
-use simplelog::*;
-use std::fs::File;
-
-mod cli;
 mod cmd;
+mod cli;
 
-pub type Result<T = ()> = anyhow::Result<T>;
-
-const VERSION: &str = concat!("v", crate_version!());
-const WINDOWS_ERROR_MSG: &str = "Windows is not supported at this time";
-
-fn main() {
-    CombinedLogger::init(vec![
-        TermLogger::new(
-            LevelFilter::Info,
-            Config::default(),
-            TerminalMode::Mixed,
-            ColorChoice::Auto,
-        ),
-        WriteLogger::new(
-            LevelFilter::Info,
-            Config::default(),
-            File::create("tembo.log").unwrap(),
-        ),
-    ])
-    .unwrap();
-
-    // Windows is not supported currently, will alert user and stop process immediately
-    if cfg!(windows) {
-        warn!("{}", crate::WINDOWS_ERROR_MSG);
-
-        std::process::exit(101);
-    }
-
-    let command = create_clap_command();
-    let matches = command.get_matches();
-
-    let res = match matches.subcommand() {
-        Some(("init", sub_matches)) => cmd::init::execute(sub_matches),
-        Some(("context", sub_matches)) => cmd::context::execute(sub_matches),
-        Some(("validate", sub_matches)) => cmd::validate::execute(sub_matches),
-        Some(("apply", sub_matches)) => cmd::apply::execute(sub_matches),
-        Some(("delete", sub_matches)) => cmd::delete::execute(sub_matches),
-        Some(("completions", sub_matches)) => (|| {
-            let shell = sub_matches
-                .get_one::<Shell>("shell")
-                .with_context(|| "Shell name missing.")?;
+#[derive(Parser)]
+#[clap(author = crate_authors!("\n"), version = crate_version!(), about = "Tembo CLI", long_about = None)]
+struct App {
+    #[clap(flatten)]
+    global_opts: GlobalOpts,
```

Comment:
There were some issues with input validation of commands in subcommands, for example showing help if you have a non-existing subcommand of a subcommand. So I've tidied this code to match the subcommands example in the docs here

https://docs.rs/clap/latest/clap/_derive/_tutorial/chapter_2/index.html#subcommands

Reply:
For example, `tembo context --help` will now show:

```
Manage Tembo contexts

Usage: tembo context <COMMAND>

Commands:
  list  List all available contexts
  set   Set the current context
  help  Print this message or the help of the given subcommand(s)

Options:
  -h, --help  Print help
```

Reply:
Awesome! Thank you.